infrastructure/repositories/base_keycloak_repository.go

```diff
@@ -0,0 +1,157 @@
+package repositories
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/PTSS-Support/identity-service/infrastructure/config"
+	"github.com/PTSS-Support/identity-service/infrastructure/util"
+)
+
+// BaseKeycloakRepository contains common functionality for Keycloak repositories
+type BaseKeycloakRepository struct {
+	config     *config.KeycloakConfig
+	httpClient *http.Client
+	logger     util.Logger
+}
+
+// NewBaseKeycloakRepository creates a new base repository instance
+func NewBaseKeycloakRepository(config *config.KeycloakConfig) *BaseKeycloakRepository {
+	return &BaseKeycloakRepository{
+		config:     config,
+		httpClient: &http.Client{},
+		logger:     util.NewLogger("BaseKeycloakRepository"),
+	}
+}
+
+// makeRequest is a helper method for making HTTP requests
+func (r *BaseKeycloakRepository) makeRequest(ctx context.Context, method, url string, data url.Values) (*http.Response, error) {
+	log := r.logger.WithContext(ctx)
+	log.Debug("Making HTTP request",
+		"method", method,
+		"url", url,
+		"data", data.Encode())
+
+	req, err := http.NewRequestWithContext(ctx, method, url, strings.NewReader(data.Encode()))
+	if err != nil {
+		log.Error("Failed to create HTTP request", "error", err)
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := r.httpClient.Do(req)
+	if err != nil {
+		log.Error("HTTP request failed", "error", err)
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+
+	log.Debug("Received HTTP response",
+		"statusCode", resp.StatusCode,
+		"headers", resp.Header)
+
+	return resp, nil
+}
+
+// getAdminToken retrieves an admin token from Keycloak
```

Comment:
this comment is quite unnecessary, method speaks for itself.

Reply:
Yesss tru